migration_guides/upgrading_to_terragrunt_0.12.x.md

```diff
@@ -0,0 +1,174 @@
+# Upgrading to Terragrunt 0.12.x
+
+## Background
+
+Terragrunt was originally created to support two features that were not available in Terraform: defining remote state
+configuration in a file (rather than via CLI commands) and locking. As of version `0.9.0`, Terraform now supports both of
+these features natively, so we have made some changes to Terragrunt:
+
+1. Terragrunt still supports remote state configuration, so you can take advantage of Terragrunt's interpolation
+   functions.
+1. Terragrunt no longer supports locking.
+
+
+## Migration guide
+
+The following sections outlines the steps you may need to take in order to migrate from Terragrunt `<= v0.11.x`
+to Terragrunt `0.12.x`. If you are using Terraform `<= 0.8.x`, see the upgrade guide on the Terraform website
+[Upgrading to Terraform v0.9](https://www.terraform.io/upgrade-guides/0-9.html).
```

Comment:
Version numbers are not typically included in the backticks. Plain text is fine for those.